src/qobj/states.jl

```diff
@@ -54,3 +121,151 @@ function rand_dm(N::Integer; dims::Vector{Int} = [N])
     ρ /= tr(ρ)
     return QuantumObject(ρ; type = Operator, dims = dims)
 end
+
+@doc raw"""
+    spin_state(j::Real, m::Real)
+
+Generate the spin state: ``|j, m\rangle``
+
+The eigenstate of the Spin-`j` ``S_z`` operator with eigenvalue `m`, where where `j` is the spin quantum number and can be a non-negative integer or half-integer
+
+See also [`jmat`](@ref).
+"""
+function spin_state(j::Real, m::Real)
+    J = 2 * j + 1
+    ((floor(J) != J) || (j < 0)) &&
+        throw(ArgumentError("The spin quantum number (j) must be a non-negative integer or half-integer."))
+
+    Δ = j - m
+    ((floor(Δ) != Δ) || (Δ < 0)) &&
+        throw(ArgumentError("Invalid eigenvalue m: (j - m) must be a non-negative integer."))
+    (m < (-j)) && throw(ArgumentError("Invalid eigenvalue m, must satisfy: -j ≤ m ≤ j"))
+
+    return basis(Int(J), Int(Δ))
+end
+
+@doc raw"""
+    spin_coherent(j::Real, θ::Real, ϕ::Real)
+
+Generate the coherent spin state (rotation of the ``|j, j\rangle`` state), namely
+
+```math
+|\theta, \phi \rangle = R(\theta, \phi) |j, j\rangle
+```
+
+where the rotation operator is defined as
+
+```math
+R(\theta, \phi) = \exp \left( \frac{\theta}{2} (S_- e^{i\phi} - S_+ e^{-i\phi}) \right)
+```
+
+# Arguments
+- `j::Real`: The spin quantum number and can be a non-negative integer or half-integer
+- `θ::Real`: rotation angle from z-axis
+- `ϕ::Real`: rotation angle from x-axis
+
+See also [`jmat`](@ref) and [`spin_state`](@ref).
+
+# Reference
+- [Robert Jones, Spin Coherent States and Statistical Physics](https://web.mit.edu/8.334/www/grades/projects/projects19/JonesRobert.pdf)
+"""
+function spin_coherent(j::Real, θ::Real, ϕ::Real)
+    Sm = jmat(j, Val(:-))
+    return exp(0.5 * θ * (Sm * exp(1im * ϕ) - Sm' * exp(-1im * ϕ))) * spin_state(j, j)
+end
+
+@doc raw"""
+    bell_state(state::String="00")
+
+Return the corresponding Bell state:
+- `"00"`: ``( |00\rangle + |11\rangle ) / \sqrt{2}``
+- `"01"`: ``( |00\rangle - |11\rangle ) / \sqrt{2}``
+- `"10"`: ``( |01\rangle + |10\rangle ) / \sqrt{2}``
+- `"11"`: ``( |01\rangle - |10\rangle ) / \sqrt{2}``
+
+# Example
+
+```
+julia> bell_state("00")
+Quantum Object:   type=Ket   dims=[2, 2]   size=(4,)
+4-element Vector{ComplexF64}:
+ 0.7071067811865475 + 0.0im
+                0.0 + 0.0im
+                0.0 + 0.0im
+ 0.7071067811865475 + 0.0im
+```
+"""
+function bell_state(state::String = "00")
```

Comment:
Why do we have a string as an argument? I think it is better to have two integer arguments.

Reply:
okay